File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sortByBits(int* arr, int arrSize, int* returnSize) {
    int cd(int num){
        int c=0;
        while(num!=0){
            num&=(num-1);
            c++;}
        return c;
    }
    *returnSize=arrSize;

    int compare(const void* a,const void* b){
        int ca=cd(*(int*)a),cb=cd(*(int*)b);
        if(ca!=cb) return ca-cb;
        else return *(int*)a-*(int*)b;
    }

    qsort(arr,arrSize,sizeof(int),compare);

    return arr;
    
}

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.java
class Solution {
    public int[] sortByBits(int[] arr) {
        int len=arr.length;
        for(int i=0;i<len;i++){
            for(int j=i+1;j<len;j++){
                int bit1=counts(arr[i]),bit2=counts(arr[j]);
                if((bit1>bit2)||(bit1==bit2&&arr[i]>arr[j])){
                    int t=arr[j];
                    arr[j]=arr[i];
                    arr[i]=t;
                }
            }
        }
        return arr;
    }

    public static int counts(int n){
        int c=0;
        while(n>0){
            if((n&1)==1)c++;
            n>>=1;
        }
        return c;
    }
}